File: portfolio/projecteuler-perl/009.pl
#perl

=pod

project euler problem 9

A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,
a2 + b2 = c2

For example, 32 + 42 = 9 + 16 = 25 = 52.

There exists exactly one Pythagorean triplet for which a + b + c = 1000.
Find the product abc.

=cut

use warnings;
use strict;

my $i=0;

sub solve{foreach my $a (2..$_[0]-2){foreach my $b (2..$_[0]-$a){
            $i++;
            my $c = 1000-$a-$b;
            return ($a*$b*$c) if ($a**2+$b**2==$c**2);
        }
    }
}

print "\n\tAnswer: ", solve(1000), ", found after $i attempts.\n";

File: portfolio/projecteuler-python/009.py

# ProjectEuler - Problem #9
# solved using Python by Michael Hillendahl

"""
A Pythagorean triplet is a set of three natural numbers,a < b < c, for which,

a2 + b2 = c2
For example, 3^2 + 4^2 = 9 + 16 = 25 = 5^2.

There exists exactly one Pythagorean triplet for which a + b + c = 1000.
Find the product abc.
"""

# SOLUTION
import math

def solve(target):
    for a in range(1,target):
        for b in range(1,target-a):
            c = 1000 - a - b
            if c == math.sqrt(a**2+b**2) : return a*b*c
            
print solve(1000)